### ISSUE TITLE
WebAssembly.Memory(): could not allocate memory

### ISSUE THREAD

**Issue by hristijankiko** [naptha/tesseract.js#476]
**Describe the bug**
Tesseract.js is throwing memory errors while workers are being initialized. I don't think think should be an issue since the device has 4 GB RAM and initializing the workers in sequence instead of in parallel yields the same result. 

Is there a way I can determine what is the root cause of this issue?

**To Reproduce**
Try to initialize more than 2 workers on mobile device with 8 cores and a value of 8 in window.navigator.hardwareConcurrency.

**Expected behavior**
I assume I should be able to initialize and use at least 50% of the hardwareConcurrency value as workers.

**Screenshots**
![image](https://user-images.githubusercontent.com/8568277/90469277-bdc06e80-e15b-11ea-8b5f-b28090dcfebe.png)

**Smartphone (please complete the following information):**
 - Device: Samsung Galaxy S9
 - OS: Android 10
 - Browser: Chrome
 - Version: 84.0.4147.125

**Comment by oke007** [comment]
i am also facing error... uncaught (in promise) rangeerror: webassembly.memory(): could not allocate memory

When loading the OCR page second time or more.... Couldn't find any information on Google so i came here to discuss. is any solution for this?

**Comment by hristijankiko** [comment]
One of the causes is that Chromium does not free memory on page reload if DevTools are open

https://bugs.chromium.org/p/chromium/issues/detail?id=935169&can=2&start=0&num=100&q=&colspec=ID%20Pri%20M%20Stars%20ReleaseBlock%20Component%20Status%20Owner%20Summary%20OS%20Modified&groupby=&sort=

**Comment by oke007** [comment]
Means if i use my ionic application in production environment instead of dev environment it may solve the issue for memory free on page reload?

**Comment by fr3shn3ss** [comment]
@oke007 For me, I get the Memory error on the testing environment but not the prod environment. My prod runs on an android phone

**Comment by Likuichao** [comment]
anyone solve？

**Comment by fr3shn3ss** [comment]
@Likuichao I realized that this happens when the workers aren't terminated correctly. Some browsers, don't terminate the workers even if you refresh the page. Closing the tab will terminate workers correctly or you should terminate all workers using terminate() function

**Comment by Balearica** [comment]
I believe the memory issues are significantly improved (if not resolved) in Tesseract.js v3.  If this issue is still active, please confirm whether your issue remains using the latest version.

**Comment by Balearica** [comment]
Closing as stale.  Feel free to reopen or open a new issue if a reasonably sized OCR task causes memory-related crashes in the current version (v3).
